lms/djangoapps/grades/models.py

```diff
@@ -436,4 +442,7 @@ def update_or_create_course_grade(cls, user_id, course_id, **kwargs):
             course_id=course_id,
             defaults=kwargs
         )
+        if grade.letter_grade and not grade.date_passed:
```

Comment:
Looking through the `passed` logic, it seems more foolproof than what we have here. I actually didn't know that existed until now. I would use that, all else being equal.